spec/cocina/models/collection_shared_examples.rb

```diff
@@ -0,0 +1,290 @@
+
+require 'spec_helper'
+
+# These shared_examples are meant to be used by Collection and RequestCollection specs in
+# order to de-dup test code for all the functionality they have in common.
+# The caller must define required_properties as a hash containing
+#   the minimal required properties that must be provided to (Request)Collection.new
+RSpec.shared_examples 'it has collection attributes' do
+  let(:instance) { described_class.new(properties) }
+  let(:collection_type) { Cocina::Models::Vocab.collection }
+  # see block comment for info about required_properties
+  let(:properties) { required_properties }
+
+  describe 'initialization' do
+    context 'with minimal required properties provided' do
+      it 'populates required attributes passed in' do
+        if required_properties[:externalIdentifier]
+          expect(instance.externalIdentifier).to eq required_properties[:externalIdentifier]
+        end
+        expect(instance.label).to eq required_properties[:label]
+        expect(instance.type).to eq required_properties[:type]
+        expect(instance.version).to eq required_properties[:version]
+      end
+
+      it 'populates non-passed required attributes with default values' do
+        expect(instance.access).to be_kind_of(Cocina::Models::Collection::Access)
+        expect(instance.access.access).to eq 'dark'
+
+        expect(instance.administrative).to be_kind_of(Cocina::Models::Collection::Administrative)
+        expect(instance.administrative.releaseTags).to eq []
+        expect(instance.administrative.hasAdminPolicy).to be_nil
+
+        expect(instance.identification).to be_kind_of(Cocina::Models::Collection::Identification)
+        expect(instance.identification.catalogLinks).to be_nil
+
+        expect(instance.structural).to be_kind_of(Cocina::Models::Collection::Structural)
+        expect(instance.structural.attributes.size).to eq 0
+      end
+    end
+
+    context 'with a string version property' do
+      let(:properties) { required_properties.merge(version: required_properties[:version].to_s) }
+
+      it 'coerces to integer' do
+        expect(instance.version).to eq required_properties[:version]
+      end
+    end
+
+    context 'with all specifiable properties' do
+      let(:properties) do
+        required_properties.merge(
+          access: {
+            access: 'stanford'
+          },
+          administrative: {
+            hasAdminPolicy: 'druid:mx123cd4567',
+            releaseTags: [
+              {
+                who: 'Justin',
+                what: 'collection',
+                date: '2018-11-23T00:44:52Z',
+                to: 'Searchworks',
+                release: 'true'
+              },
+              {
+                who: 'Other Justin',
+                what: 'self',
+                date: '2017-10-20T15:42:15Z',
+                to: 'Earthworks',
+                release: 'false'
+              }
+            ]
+          },
+          description: {
```

Comment:
I removed sourceId attribute as it's not in the struct and I'm not sure we associate collection objects with a sourceId ?